### PR TITLE
Miscellaneous Changes

### PR DESCRIPTION
**Pull request by tarekgh**
-	Rename OpenTelemetry. DistributedContext namespace to Context. The reason is there is added class DistributedContext with the same name.
-	Fix some misspelling in the Gauge names.
-	Added the overloads in the Counter, Gauge and Measure classes which takes DistributedContext parameters.
-	Replaced CounterHandle, GaugeHandle, and MeasureHandle structs by just one abstract class MetricsHandle. As allowing implementing abstract counter classes, we should allow the same for the handle classes.

### CODE REVIEW

**Comment by tarekgh** [comment]
@SergeyKanzhelev @cijothomas could you please have a look? Thanks.

**Comment by tarekgh** [comment]
I signed it

**Comment by tarekgh** [comment]
("I signed it")

**Comment by tarekgh** [comment]
"I signed it"

**Comment by tarekgh** [comment]
I signed it

**Comment by tarekgh** [comment]
@SergeyKanzhelev @cijothomas did you have a chance to look at this one?

File: src/OpenTelemetry.Api/Metrics/MetricsHandle.cs [hunk 1]
```diff
@@ -14,23 +14,33 @@
...
     {
         /// <summary>
-        /// Adds or Increments the value of the counter handle.
+        /// Add, Set or Record the value of the counter handle.
         /// </summary>
         /// <param name="context">the associated span context.</param>
-        /// <param name="value">value by which the counter handle should be incremented.</param>
-        public void Add(in SpanContext context, T value)
+        /// <param name="value">value by which the metrics instrument handle should be updated.</param>
+        public void Update(in SpanContext context, T value)
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Api/Metrics/MetricsHandle.cs]: I don't think this is a good ideas as semantic is lost here. It is important to note for people whether to pass "increment" or current value. For me it looks like overgeneralization.

File: src/OpenTelemetry.Api/Metrics/Gauge.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿// <copyright file="Guage.cs" company="OpenTelemetry Authors">
+﻿// <copyright file="Gauge.cs" company="OpenTelemetry Authors">
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Api/Metrics/Gauge.cs]: good catch!

File: samples/Exporters/TestApplicationInsights.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@
 using System.Collections.Generic;
 using System.Threading;
 using Microsoft.ApplicationInsights.Extensibility;
-using OpenTelemetry.DistributedContext;
+using OpenTelemetry.Context;
```
Comment (SergeyKanzhelev) [inline samples/Exporters/TestApplicationInsights.cs]: makes sense.

**Comment by tarekgh** [comment]
> I feel that MeasureHandle is overgeneralization. We have this in Application Insights and it is confusing for API users. Everything else looks good

I see your point, Usually users will do the following: 

```C#
MetricsHandle counterHandle = Meter.Default.GetLongCounter();
counterHandle.Update();
```

which is kind of acceptable for me :-) 

I'll think about this one more and get back to you.

**Comment by tarekgh** [comment]
@SergeyKanzhelev I have returned back the handle classes. please have a look and let me know if you have any more feedback.

**Comment by SergeyKanzhelev** [comment]
have you signed CLA with the same e-mail as used in commits? I'll try to close and re-open this PR now to see if it will help.

**Comment by SergeyKanzhelev** [comment]
can you please try to re-base so there will be no merge commits... maybe it will help

**Comment by tarekgh** [comment]
> can you please try to re-base so there will be no merge commits... maybe it will help

I'll try to do that soon.

**Comment by tarekgh** [comment]
@SergeyKanzhelev everything is green now :-)

File: src/OpenTelemetry.Api/Metrics/GaugeHandle.cs [hunk 4]
```diff
@@ -14,27 +14,30 @@
...
     /// </summary>
-    /// <typeparam name="T">The type of the Gauge. Only long and double are supported now.</typeparam>
-    public struct GuageHandle<T>
+    /// <typeparam name="T">The type of gauge. Only long and double are supported now.</typeparam>
+    public abstract class GaugeHandle<T>
         where T : struct
     {
         /// <summary>
-        /// Sets the value of the guague.
+        /// Add, Set or Record the value of the counter handle.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Metrics/GaugeHandle.cs]: you can restore old comment - Sets the value of the gauge handle.

File: src/OpenTelemetry.Api/Metrics/GaugeHandle.cs [hunk 5]
```diff
@@ -14,27 +14,30 @@
...
         /// </summary>
+        /// <param name="context">the associated span context.</param>
+        /// <param name="value">value by which the gauge handle should be set.</param>
+        public abstract void Set(in SpanContext context, T value);
+
         /// <summary>
-        /// Sets the value of the guague handle.
+        /// Add, Set or Record the value of the counter handle.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Metrics/GaugeHandle.cs]: you can restore old comment - Sets the value of the gauge handle.

File: src/OpenTelemetry.Api/Metrics/MeasureHandle.cs [hunk 6]
```diff
@@ -14,27 +14,30 @@
...
-        /// Records a measure.
+        /// Record the value of the Measure handle.
         /// </summary>
+        /// <param name="context">the associated span context.</param>
+        /// <param name="value">value by which the Measure handle should be Recorded.</param>
+        public abstract void Record(in SpanContext context, T value);
+
         /// <summary>
-        /// Sets the value of the guague handle.
+        /// Add, Set or Record the value of the counter handle.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Metrics/MeasureHandle.cs]: comment change to just 'Record the value of the measure handle'.

File: src/OpenTelemetry.Api/Metrics/GaugeHandle.cs (hunk 4 above)
Comment (SergeyKanzhelev) [reply src/OpenTelemetry.Api/Metrics/GaugeHandle.cs]:
```suggestion
        /// Sets the value of the guague.
```

File: src/OpenTelemetry.Api/Metrics/GaugeHandle.cs (hunk 5 above)
Comment (SergeyKanzhelev) [reply src/OpenTelemetry.Api/Metrics/GaugeHandle.cs]:
```suggestion
        /// Sets the value of the guague handle.
```

File: src/OpenTelemetry.Api/Metrics/MeasureHandle.cs (hunk 6 above)
Comment (SergeyKanzhelev) [reply src/OpenTelemetry.Api/Metrics/MeasureHandle.cs]:
```suggestion
        /// Sets the value of the guague handle.
```

**Comment by SergeyKanzhelev** [comment]
@tarekgh would you mind to revert comments. Just click on "commit suggestion" on each. And I'll merge than

**Comment by tarekgh** [comment]
@SergeyKanzhelev done!

**Pull request merged**